sph/include/sph/cooling.hpp

```diff
@@ -0,0 +1,141 @@
+//
+// Created by Noah Kubli on 28.02.22.
+//
+
+#ifndef SPHEXA_COOLING_HPP
+#define SPHEXA_COOLING_HPP
+
+#define CONFIG_BFLOAT_8
+extern "C"
+{
+#include <grackle.h>
+}
+#include <cmath>
+static constexpr double m_sun = 1.989e33;
+static constexpr double pc    = 3.086e18;
+static constexpr double G_cgs = 6.674e-8;
+
+static code_units grackle_units;
+
+template<typename T, typename Dataset>
+void cool_particle(Dataset& d, size_t i)
```

Comment:
This function receives way too much information compared to what it is actually using.
The arguments contain information of _all_ particle fields for _all_ particles, even though
this is only using the composite fractions and rho and u for a _single_ particle.
I would propose you pass here only the pointers to rho[i], u[i],HI[i], etc.

Reply:
Yes, I will change that.